core/src/test/scala/unit/kafka/log/LogCleanerTest.scala

```diff
@@ -642,17 +642,17 @@ class LogCleanerTest extends JUnitSuite {
     }
 
     // grouping by very large values should result in a single group with all the segments in it
-    var groups = cleaner.groupSegmentsBySize(log.logSegments, maxSize = Int.MaxValue, maxIndexSize = Int.MaxValue)
+    var groups = cleaner.groupSegmentsBySize(log.logSegments, maxSize = Int.MaxValue, maxIndexSize = Int.MaxValue, log.logSegments.last.index.lastOffset + 1)
```

Comment:
We can use log.logEndOffset instead of log.logSegments.last.index.lastOffset + 1. The former is much cheaper. Ditto in a few other places below.